src/app/signOut/signOut.component.js

```diff
@@ -25,20 +26,15 @@ class SignOutController {
   }
 
   redirectToLocationPriorToSignOut () {
-    this.showRedirectingLoadingIcon = true
-    const locationToReturnUser = this.sessionService.getLocationOnLogin()
+    this.redirectToPreviousLocation = true
+    const locationToReturnUser = this.sessionService.getStoredLocation()
     if (locationToReturnUser) {
-      this.sessionService.removeLocationOnLogin()
+      this.sessionService.removeStoredLocation()
       this.$window.location.href = locationToReturnUser
     } else {
       this.redirectToHomepage()
```

Comment:
Wouldn't `this.redirectToPreviousLocation` need to be `false` in this branch? Maybe you can move `this.redirectToPreviousLocation = true` inside the `if`.

Reply:
Good catch

Reply:
It should be fixed. Let me check

Reply:
Yes, it is fixed. `this.redirectToPreviousLocation` is only set to `TRUE` when  `locationToReturnUser` is truthy.

Reply:
Sorry, I clicked the wrong thing and I was looking at the old version of that method. Looks great now!